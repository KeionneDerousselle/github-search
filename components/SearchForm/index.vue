<template>
  <form-validator
    class="search__form"
    @submit="handleSearchSubmitted">
    <template #default="{ invalid }">
      <div class="search__form__container">
        <searchfield
          id="search-box"
          v-model="searchTerm"
          class="search__box"
          placeholder="Search"
          rules="required"
          label="Search Term"
          name="Search Term"
          vee-validate-name="A search term"
          mode="aggressive"
          immediate
          :show-errors="false"
          :calc-input-classes="getSearchInputClasses"
          :calc-container-classes="getSearchInputContainerClasses" />

        <kd-github-search-button
          id="search-button"
          class="search__button"
          type="submit"
          :disabled="invalid || performingSearch"
          :loading="performingSearch">
          <span class="sr-only">Submit</span>
          <check-icon
            v-show="!performingSearch"
            class="search__button__icon"
            aria-hidden="true" />
        </kd-github-search-button>
      </div>
    </template>
  </form-validator>
</template>
<script>
import CheckIcon from '@/components/Icons/Check'
import Button from '@/components/Button'
import FormValidator from '@/components/FormValidator'
import Searchfield from '@/components/Searchfield'

import { mapActions, mapGetters } from 'vuex'

export default {
  components: {
    CheckIcon,
    FormValidator,
    Searchfield,
    'kd-github-search-button': Button
  },

  data: () => ({
    searchTerm: '',
    performingSearch: false,
    fetchingTheNextPage: false
  }),

  computed: {
    ...mapGetters('users', [
      'currentSearchTerm',
      'resultsPerPage'
    ])
  },

  methods: {
    ...mapActions('users', [
      'setSearchTerm',
      'search'
    ]),

    getSearchInputContainerClasses(errors, isFocused) {
      const classStrats = [
        {
          shouldApply: () => isFocused,
          classes: [ 'search__container', 'search__container--focused' ]
        },
        {
          shouldApply: () => true,
          classes: ['search__container']
        }
      ]

      return classStrats.find(cs => cs.shouldApply()).classes
    },

    getSearchInputClasses(errors, isFocused) {
      // const anyErrors = errors && errors.length > 0

      const classStrats = [
        {
          shouldApply: () => true,
          classes: ['search__input']
        }
      ]

      return classStrats.find(cs => cs.shouldApply()).classes
    },

    handleSearchSubmitted() {
      // TODO: Loading State
      // TODO: Handle Errors

      if (this.searchTerm && (this.searchTerm.toLowerCase() !== this.currentSearchTerm.toLowerCase())) {
        this.performingSearch = true

        return this.search({ searchTerm: this.searchTerm, page: 1, resultsPerPage: this.resultsPerPage })
          .then(() => this.setSearchTerm(this.searchTerm))
          .catch(console.error)
          .finally(() => {
            this.performingSearch = false
          })
      }
    }
  }
}
</script>
<style lang="scss">
.search__form__container {
  @apply flex items-start justify-between flex-none;
}

.search__box {
  @apply flex-grow w-11/12;
  @apply mr-3;
  @apply md:mr-4;
}

.search__container {
  @apply shadow-lg bg-indigo-500 text-white;

  &--focused {
    @apply ring-4 ring-indigo-400;
  }
}

.search__input {
  @apply placeholder-indigo-50;

  &:-webkit-autofill,
  &:-webkit-autofill:hover,
  &:-webkit-autofill:focus,
  &:-webkit-autofill:active,
  &:-internal-autofill-selected {
    @apply bg-indigo-500 text-white;

    -webkit-text-fill-color: white;
    caret-color: white;
  }
}

.search__button {
  @apply shadow-lg border-0 bg-indigo-200 flex-none text-indigo-800 h-full;

  &:hover:not(:disabled) {
    @apply bg-indigo-300 shadow-2xl;
  }

  &:focus:not(:active) {
    @apply shadow-2xl ring-indigo-400 ring-4;
  }

  &:disabled {
    @apply bg-gray-300;
  }

  &:disabled .search__button__icon {
    @apply text-gray-600;
  }

  &__icon {
    @apply font-bold w-5 h-5;
  }
}
</style>
