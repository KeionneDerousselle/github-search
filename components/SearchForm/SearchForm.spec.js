import flushPromises from 'flush-promises'
import { extend } from 'vee-validate'
import { required } from 'vee-validate/dist/rules'
import SearchForm from '.'

describe('Search Form', () => {
  let wrapper

  beforeAll(() => {
    extend('required', required)
    jest.useFakeTimers()
  })

  afterAll(() => {
    jest.useRealTimers()
  })

  describe('when a search is performed without a search term', () => {
    let searchButton
    let mockSearchAction

    beforeAll(async () => {
      mockSearchAction = jest.fn().mockResolvedValue()

      wrapper = mountPreMocked(SearchForm, {
        store: {
          users: {
            getters: {
              currentSearchTerm: jest.fn().mockReturnValue(''),
              resultsPerPage: jest.fn().mockReturnValue(25)
            },
            actions: {
              setSearchTerm: jest.fn().mockResolvedValue(),
              search: mockSearchAction
            }
          }
        }
      })

      searchButton = wrapper.get('#search-button')

      searchButton.trigger('click')
      await flushValidationUpdates(wrapper)
    })

    afterAll(() => {
      mockSearchAction.mockRestore()
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should disable the search submit button', () => {
      expect(searchButton.attributes().disabled).toBeTruthy()
    })

    it('should not call the search action', () => {
      expect(mockSearchAction).not.toHaveBeenCalled()
    })
  })

  describe('when an invalid search is performed', () => {
    let searchButton
    let mockSearchAction

    // TODO: include validation rules
    beforeAll(async () => {
      mockSearchAction = jest.fn().mockResolvedValue()

      wrapper = mountPreMocked(SearchForm, {
        store: {
          users: {
            getters: {
              currentSearchTerm: jest.fn().mockReturnValue(''),
              resultsPerPage: jest.fn().mockReturnValue(25)
            },
            actions: {
              setSearchTerm: jest.fn().mockResolvedValue(),
              search: mockSearchAction
            }
          }
        }
      })

      const searchBox = wrapper.get('#search-box')

      searchButton = wrapper.get('#search-button')

      searchBox.setValue('')
      searchBox.trigger('blur')
      searchButton.trigger('click')

      await flushValidationUpdates(wrapper)
    })

    afterAll(() => {
      mockSearchAction.mockRestore()
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should disable the search submit button', () => {
      expect(searchButton.attributes().disabled).toBeTruthy()
    })

    it('should not call the search action', () => {
      expect(mockSearchAction).not.toHaveBeenCalled()
    })
  })

  describe('when a valid search is performed', () => {
    let searchButton
    let mockSearchAction
    let mockSetSearchTermAction
    let searchTerm
    let page
    let resultsPerPage

    // TODO: include validation rules
    beforeAll(async () => {
      searchTerm = 'test'
      mockSearchAction = jest.fn().mockResolvedValue()
      mockSetSearchTermAction = jest.fn().mockResolvedValue()
      page = 1
      resultsPerPage = 25

      wrapper = mountPreMocked(SearchForm, {
        store: {
          users: {
            getters: {
              currentSearchTerm: jest.fn().mockReturnValue(''),
              resultsPerPage: jest.fn().mockReturnValue(resultsPerPage)
            },
            actions: {
              setSearchTerm: mockSetSearchTermAction,
              search: mockSearchAction
            }
          }
        }
      })

      const searchBox = wrapper.get('#search-box')

      searchButton = wrapper.get('#search-button')

      searchBox.setValue(searchTerm)
      searchBox.trigger('blur')
      await flushValidationUpdates(wrapper)

      searchButton.trigger('click')
      await flushValidationUpdates(wrapper)
    })

    afterAll(() => {
      mockSearchAction.mockRestore()
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should not disable the search submit button', () => {
      expect(searchButton.attributes().disabled).toBeFalsy()
    })

    it('should set the current search term in the store', () => {
      expect(mockSetSearchTermAction).toHaveBeenCalledWith(expect.any(Object), searchTerm)
    })

    it('should call the search action', () => {
      expect(mockSearchAction).toHaveBeenCalledWith(expect.any(Object), { searchTerm, page, resultsPerPage })
    })
  })

  describe('when a search is performed that matches the current search term', () => {
    let searchButton
    let mockSearchAction
    let mockSetSearchTermAction
    let searchTerm

    // TODO: include validation rules
    beforeAll(async () => {
      searchTerm = 'test'
      mockSearchAction = jest.fn().mockResolvedValue([])
      mockSetSearchTermAction = jest.fn().mockResolvedValue()

      wrapper = mountPreMocked(SearchForm, {
        store: {
          users: {
            getters: {
              currentSearchTerm: jest.fn().mockReturnValue('test'),
              resultsPerPage: jest.fn().mockReturnValue(25)
            },
            actions: {
              setSearchTerm: mockSetSearchTermAction,
              search: mockSearchAction
            }
          }
        }
      })

      const searchBox = wrapper.get('#search-box')

      searchButton = wrapper.get('#search-button')

      searchBox.setValue(searchTerm)
      searchBox.trigger('blur')
      await flushValidationUpdates(wrapper)

      searchButton.trigger('click')
      await flushValidationUpdates(wrapper)
    })

    afterAll(() => {
      mockSearchAction.mockRestore()
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should not disable the search submit button', () => {
      expect(searchButton.attributes().disabled).toBeFalsy()
    })

    it('should not set the current search term in the store', () => {
      expect(mockSetSearchTermAction).not.toHaveBeenCalled()
    })

    it('should not call the search action', () => {
      expect(mockSearchAction).not.toHaveBeenCalled()
    })
  })

  describe('when the search term container is focused', () => {
    let searchContainerClasses

    beforeAll(async () => {
      wrapper = mountPreMocked(SearchForm, {
        store: {
          users: {
            getters: {
              currentSearchTerm: jest.fn().mockReturnValue(''),
              resultsPerPage: jest.fn().mockReturnValue(25)
            },
            actions: {
              setSearchTerm: jest.fn().mockResolvedValue(),
              search: jest.fn().mockResolvedValue()
            }
          }
        }
      })

      const searchContainer = wrapper.get('.searchfield')

      searchContainer.trigger('focus')
      await flushPromises()
      await wrapper.vm.$nextTick()

      searchContainerClasses = searchContainer.classes()
    })

    afterAll(() => {
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should contain the default search container css classes', () => {
      expect(searchContainerClasses).toContain('search__container')
    })

    it('should contain the focused search container css classes', () => {
      expect(searchContainerClasses).toContain('search__container')
      expect(searchContainerClasses).toContain('search__container--focused')
    })
  })

  describe('when the search term container is not focused and there are no search errors', () => {
    let searchContainerClasses

    beforeAll(() => {
      wrapper = mountPreMocked(SearchForm, {
        store: {
          users: {
            getters: {
              currentSearchTerm: jest.fn().mockReturnValue(''),
              resultsPerPage: jest.fn().mockReturnValue(25)
            },
            actions: {
              setSearchTerm: jest.fn().mockResolvedValue(),
              search: jest.fn().mockResolvedValue()
            }
          }
        }
      })

      searchContainerClasses = wrapper.get('.searchfield').classes()
    })

    afterAll(() => {
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should contain the default search container css classes', () => {
      expect(searchContainerClasses).toContain('search__container')
    })

    it('should not contain the focused search container css classes', () => {
      expect(searchContainerClasses).not.toContain('search__container--focused')
    })
  })

  describe('when the search term input is not focused and there are no errors', () => {
    let searchInputClasses

    beforeAll(() => {
      wrapper = mountPreMocked(SearchForm, {
        store: {
          users: {
            getters: {
              currentSearchTerm: jest.fn().mockReturnValue(''),
              resultsPerPage: jest.fn().mockReturnValue(25)
            },
            actions: {
              setSearchTerm: jest.fn().mockResolvedValue(),
              search: jest.fn().mockResolvedValue()
            }
          }
        }
      })

      searchInputClasses = wrapper.get('#search-box').classes()
    })

    afterAll(() => {
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should contain the default search input css classes', () => {
      expect(searchInputClasses).toContain('search__input')
    })
  })

  describe('when clicking the clear search text button', () => {
    let searchClearButton
    let searchBox

    beforeAll(async () => {
      wrapper = mountPreMocked(SearchForm, {
        store: {
          users: {
            getters: {
              currentSearchTerm: jest.fn().mockReturnValue(''),
              resultsPerPage: jest.fn().mockReturnValue(25)
            },
            actions: {
              setSearchTerm: jest.fn().mockResolvedValue(),
              search: jest.fn().mockResolvedValue()
            }
          }
        }
      })

      searchBox = wrapper.get('#search-box')
      searchBox.setValue('test')
      searchBox.trigger('blur')
      await flushValidationUpdates(wrapper)

      searchClearButton = wrapper.get('.searchfield__clear')
      searchClearButton.trigger('click')
      await flushValidationUpdates(wrapper)
    })

    afterAll(() => {
      wrapper.destroy()
    })

    it('should render as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should render the nuxt tag', () => {
      expect(wrapper.get('#search-button').attributes().disabled).toBeTruthy()
    })

    it('should set the search term to an empty string', () => {
      expect(wrapper.vm.searchTerm).toBe('')
    })

    it('should clear the search field', () => {
      expect(wrapper.get('#search-box').element.value).toBe('')
    })
  })
})
