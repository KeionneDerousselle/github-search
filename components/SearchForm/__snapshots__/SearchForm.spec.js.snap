// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form when a search is performed that matches the current search term should render as expected 1`] = `
<form class="form search__form">
  <div class="search__form__container">
    <div class="attrs-fix search__box">
      <div class="input searchfield-container"><label for="search-box" class="input__label sr-only">
          Search Term
        </label>
        <div tabindex="0" class="searchfield rounded-lg search__container"><span class="searchfield__left"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" class="searchfield__icon"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></span> <input type="search" placeholder="Search" name="Search Term" id="search-box" class="searchfield__input search__input"> <span class="searchfield__right"><button type="reset" class="searchfield__clear button" style=""><div class="button__content"><!----> <span class="button__content--center"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" class="searchfield__clear__icon"><path fill-rule="evenodd" d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z" clip-rule="evenodd"></path></svg></span>
          <!---->
        </div></button></span>
      </div>
      <!---->
      <!---->
    </div>
  </div> <button id="search-button" type="submit" class="search__button button">
    <div class="button__content">
      <!----> <span class="button__content--center"><span class="sr-only">Submit</span> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true" class="search__button__icon">
        <path fill-rule="evenodd" d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z" clip-rule="evenodd"></path>
      </svg></span>
      <!---->
    </div>
  </button></div>
</form>
`;

exports[`Search Form when a search is performed without a search term should render as expected 1`] = `
<form class="form search__form">
  <div class="search__form__container">
    <div class="attrs-fix search__box">
      <div class="input searchfield-container"><label for="search-box" class="input__label sr-only">
          Search Term
        </label>
        <div tabindex="0" class="searchfield rounded-lg search__container"><span class="searchfield__left"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" class="searchfield__icon"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></span> <input type="search" placeholder="Search" name="Search Term" id="search-box" class="searchfield__input search__input"> <span class="searchfield__right"><button type="reset" class="searchfield__clear button" style="display: none;"><div class="button__content"><!----> <span class="button__content--center"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" class="searchfield__clear__icon"><path fill-rule="evenodd" d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z" clip-rule="evenodd"></path></svg></span>
          <!---->
        </div></button></span>
      </div>
      <!---->
      <!---->
    </div>
  </div> <button id="search-button" type="submit" class="search__button button" disabled="disabled">
    <div class="button__content">
      <!----> <span class="button__content--center"><span class="sr-only">Submit</span> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true" class="search__button__icon">
        <path fill-rule="evenodd" d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z" clip-rule="evenodd"></path>
      </svg></span>
      <!---->
    </div>
  </button></div>
</form>
`;

exports[`Search Form when a valid search is performed should render as expected 1`] = `
<form class="form search__form">
  <div class="search__form__container">
    <div class="attrs-fix search__box">
      <div class="input searchfield-container"><label for="search-box" class="input__label sr-only">
          Search Term
        </label>
        <div tabindex="0" class="searchfield rounded-lg search__container"><span class="searchfield__left"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" class="searchfield__icon"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></span> <input type="search" placeholder="Search" name="Search Term" id="search-box" class="searchfield__input search__input"> <span class="searchfield__right"><button type="reset" class="searchfield__clear button" style=""><div class="button__content"><!----> <span class="button__content--center"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" class="searchfield__clear__icon"><path fill-rule="evenodd" d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z" clip-rule="evenodd"></path></svg></span>
          <!---->
        </div></button></span>
      </div>
      <!---->
      <!---->
    </div>
  </div> <button id="search-button" type="submit" class="search__button button">
    <div class="button__content">
      <!----> <span class="button__content--center"><span class="sr-only">Submit</span> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true" class="search__button__icon" style="">
        <path fill-rule="evenodd" d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z" clip-rule="evenodd"></path>
      </svg></span>
      <!---->
    </div>
  </button></div>
</form>
`;

exports[`Search Form when an invalid search is performed should render as expected 1`] = `
<form class="form search__form">
  <div class="search__form__container">
    <div class="attrs-fix search__box">
      <div class="input searchfield-container"><label for="search-box" class="input__label sr-only">
          Search Term
        </label>
        <div tabindex="0" class="searchfield rounded-lg search__container"><span class="searchfield__left"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" class="searchfield__icon"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></span> <input type="search" placeholder="Search" name="Search Term" id="search-box" class="searchfield__input search__input"> <span class="searchfield__right"><button type="reset" class="searchfield__clear button" style="display: none;"><div class="button__content"><!----> <span class="button__content--center"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" class="searchfield__clear__icon"><path fill-rule="evenodd" d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z" clip-rule="evenodd"></path></svg></span>
          <!---->
        </div></button></span>
      </div>
      <!---->
      <!---->
    </div>
  </div> <button id="search-button" type="submit" class="search__button button" disabled="disabled">
    <div class="button__content">
      <!----> <span class="button__content--center"><span class="sr-only">Submit</span> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true" class="search__button__icon">
        <path fill-rule="evenodd" d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z" clip-rule="evenodd"></path>
      </svg></span>
      <!---->
    </div>
  </button></div>
</form>
`;

exports[`Search Form when clicking the clear search text button should render as expected 1`] = `
<form class="form search__form">
  <div class="search__form__container">
    <div class="attrs-fix search__box">
      <div class="input searchfield-container"><label for="search-box" class="input__label sr-only">
          Search Term
        </label>
        <div tabindex="0" class="searchfield rounded-lg search__container search__container--focused"><span class="searchfield__left"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" class="searchfield__icon"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></span> <input type="search" placeholder="Search" name="Search Term" id="search-box" class="searchfield__input search__input"> <span class="searchfield__right"><button type="reset" class="searchfield__clear button" style="display: none;"><div class="button__content"><!----> <span class="button__content--center"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" class="searchfield__clear__icon"><path fill-rule="evenodd" d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z" clip-rule="evenodd"></path></svg></span>
          <!---->
        </div></button></span>
      </div>
      <!---->
      <!---->
    </div>
  </div> <button id="search-button" type="submit" class="search__button button" disabled="disabled">
    <div class="button__content">
      <!----> <span class="button__content--center"><span class="sr-only">Submit</span> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true" class="search__button__icon">
        <path fill-rule="evenodd" d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z" clip-rule="evenodd"></path>
      </svg></span>
      <!---->
    </div>
  </button></div>
</form>
`;

exports[`Search Form when the search term container is focused should render as expected 1`] = `
<form class="form search__form">
  <div class="search__form__container">
    <div class="attrs-fix search__box">
      <div class="input searchfield-container"><label for="search-box" class="input__label sr-only">
          Search Term
        </label>
        <div tabindex="0" class="searchfield rounded-lg search__container search__container--focused"><span class="searchfield__left"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" class="searchfield__icon"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></span> <input type="search" placeholder="Search" name="Search Term" id="search-box" class="searchfield__input search__input"> <span class="searchfield__right"><button type="reset" class="searchfield__clear button" style="display: none;"><div class="button__content"><!----> <span class="button__content--center"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" class="searchfield__clear__icon"><path fill-rule="evenodd" d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z" clip-rule="evenodd"></path></svg></span>
          <!---->
        </div></button></span>
      </div>
      <!---->
      <!---->
    </div>
  </div> <button id="search-button" type="submit" class="search__button button">
    <div class="button__content">
      <!----> <span class="button__content--center"><span class="sr-only">Submit</span> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true" class="search__button__icon">
        <path fill-rule="evenodd" d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z" clip-rule="evenodd"></path>
      </svg></span>
      <!---->
    </div>
  </button></div>
</form>
`;

exports[`Search Form when the search term container is not focused and there are no search errors should render as expected 1`] = `
<form class="form search__form">
  <div class="search__form__container">
    <div class="attrs-fix search__box">
      <div class="input searchfield-container"><label for="search-box" class="input__label sr-only">
          Search Term
        </label>
        <div tabindex="0" class="searchfield rounded-lg search__container"><span class="searchfield__left"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" class="searchfield__icon"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></span> <input type="search" placeholder="Search" name="Search Term" id="search-box" class="searchfield__input search__input"> <span class="searchfield__right"><button type="reset" class="searchfield__clear button" style="display: none;"><div class="button__content"><!----> <span class="button__content--center"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" class="searchfield__clear__icon"><path fill-rule="evenodd" d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z" clip-rule="evenodd"></path></svg></span>
          <!---->
        </div></button></span>
      </div>
      <!---->
      <!---->
    </div>
  </div> <button id="search-button" type="submit" class="search__button button">
    <div class="button__content">
      <!----> <span class="button__content--center"><span class="sr-only">Submit</span> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true" class="search__button__icon">
        <path fill-rule="evenodd" d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z" clip-rule="evenodd"></path>
      </svg></span>
      <!---->
    </div>
  </button></div>
</form>
`;

exports[`Search Form when the search term input is not focused and there are no errors should render as expected 1`] = `
<form class="form search__form">
  <div class="search__form__container">
    <div class="attrs-fix search__box">
      <div class="input searchfield-container"><label for="search-box" class="input__label sr-only">
          Search Term
        </label>
        <div tabindex="0" class="searchfield rounded-lg search__container"><span class="searchfield__left"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" class="searchfield__icon"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"></path></svg></span> <input type="search" placeholder="Search" name="Search Term" id="search-box" class="searchfield__input search__input"> <span class="searchfield__right"><button type="reset" class="searchfield__clear button" style="display: none;"><div class="button__content"><!----> <span class="button__content--center"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" class="searchfield__clear__icon"><path fill-rule="evenodd" d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z" clip-rule="evenodd"></path></svg></span>
          <!---->
        </div></button></span>
      </div>
      <!---->
      <!---->
    </div>
  </div> <button id="search-button" type="submit" class="search__button button">
    <div class="button__content">
      <!----> <span class="button__content--center"><span class="sr-only">Submit</span> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true" class="search__button__icon">
        <path fill-rule="evenodd" d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z" clip-rule="evenodd"></path>
      </svg></span>
      <!---->
    </div>
  </button></div>
</form>
`;
